crates/biome_configuration/src/javascript/mod.rs

```diff
@@ -100,6 +117,101 @@ impl FromStr for JsxRuntime {
     }
 }
 
+fn deserialize_optional_jsx_factory_from_string<'de, D>(
+    deserializer: D,
+) -> Result<Option<JsxFactory>, D::Error>
+where
+    D: Deserializer<'de>,
+{
+    let s = String::deserialize(deserializer)?;
+    match parse_jsx_factory(&s) {
+        Some(factory) => Ok(Some(factory)),
+        None => Err(serde::de::Error::custom(format!(
+            "expected valid identifier or qualified name, but received {s}"
+        ))),
+    }
+}
+
+fn parse_jsx_factory(value: &str) -> Option<JsxFactory> {
```

Comment:
I don't know if this is the right place to put, thinking of this is a quite general utility.